### PR TITLE
Feature/schema org for homepage

### PR DESCRIPTION
**Pull request by punalsc**
Hi @sdwilkes ,
Could you have a look at this PR.

Thanks

### CODE REVIEW

File: inc/functions-home.php [hunk 1]
```diff
@@ -419,3 +423,57 @@ function home_alert( $status, $title, $text ) {
...
+ * @since 1.0
+ */
+function render_schema(){
+    global $post;
+    $canonicalUrl = wp_get_canonical_url();
+    $pageDescription = "";
+    include_once( ABSPATH . 'wp-admin/includes/plugin.php' );
+    if ( is_plugin_active( 'wordpress-seo/wp-seo.php' ) ) {
+        if (function_exists('get_post_meta')) {
+            $pageDescription = get_post_meta($post->ID, '_yoast_wpseo_metadesc', true);
```
Comment (sdwilkes) [inline inc/functions-home.php]: Change to check if field is empty.

File: inc/functions-home.php [hunk 2]
```diff
@@ -419,3 +423,57 @@ function home_alert( $status, $title, $text ) {
...
+                     "description" : "%s"
+                     "url": "%s",
+                     "logo": "http://nationalarchives.gov.uk/wp-content/themes/tna-base/img/logo-white.png",
+                     "address": {
+                         "@type": "PostalAddress",
+                         "streetAddress": "The National Archives, Kew, Richmond, Surrey",
+                         "addressLocality": "Kew",
+                         "addressRegion": "Richmond",
+                         "postalCode": "TW9 4DU",
+                         "addressCountry": "England"
```
[the remark this answers is not in the document]
Comment (punalsc) [reply inc/functions-home.php]: done.

File: inc/functions-home.php [hunk 3]
```diff
@@ -419,3 +423,57 @@ function home_alert( $status, $title, $text ) {
...
+                     "name": "The National Archives",
+                     "legalName" : "The National Archives",
+                     "description" : "%s"
+                     "url": "%s",
+                     "logo": "http://nationalarchives.gov.uk/wp-content/themes/tna-base/img/logo-white.png",
+                     "address": {
+                         "@type": "PostalAddress",
+                         "streetAddress": "The National Archives, Kew, Richmond, Surrey",
+                         "addressLocality": "Kew",
+                         "addressRegion": "Richmond",
```
[the remark this answers is not in the document]
Comment (punalsc) [reply inc/functions-home.php]: done

File: inc/functions-home.php [hunk 4]
```diff
@@ -419,3 +423,57 @@ function home_alert( $status, $title, $text ) {
...
+    }
+    $schema =
+        '<script type="application/ld+json">
+                {
+                     "@context": "http://schema.org",
+                        "@type": "Organization",
+                        "@id": "%s",
+                     "name": "The National Archives",
+                     "legalName" : "The National Archives",
+                     "description" : "%s"
```
[the remark this answers is not in the document]
Comment (punalsc) [reply inc/functions-home.php]: done

**Comment by punalsc** [comment]
Hi @sdwilkes ,

Could you check the PR, as per your requested changes.

Thanks

**Pull request merged**